samza-core/src/main/java/org/apache/samza/clustermanager/HostAwareContainerAllocator.java

```diff
@@ -81,13 +87,27 @@ public void assignResourceRequests()  {
 
         if (expired) {
           updateExpiryMetrics(request);
-          if (resourceAvailableOnAnyHost) {
-            log.info("Request for container: {} on {} has expired. Running on ANY_HOST", request.getContainerID(), request.getPreferredHost());
-            runStreamProcessor(request, ResourceRequestState.ANY_HOST);
+
+          if (standbyContainerManager.isPresent()) {
```

Comment:
what if standby is not enabled? the current behavior is to issue any_host request. is that changed?

Reply:
Current behavior is unchanged.
This if is only triggerred if (standbyContainerManager.isPresent())
otherwise it falls into the if below, 
```else {

            if (resourceAvailableOnAnyHost) {
              log.info("Request for container: {} on {} has expired. Running on ANY_HOST", request.getContainerID(), request.getPreferredHost());
              runStreamProcessor(request, ResourceRequestState.ANY_HOST);
            } else {
              log.info("Request for container: {} on {} has expired. Requesting additional resources on ANY_HOST.", request.getContainerID(), request.getPreferredHost());
              resourceRequestState.cancelResourceRequest(request);
              requestResource(containerID, ResourceRequestState.ANY_HOST);
            }

          }```